src/backend/joanie/tests/core/api/order/test_submit_for_signature.py

```diff
@@ -27,7 +27,9 @@ def test_api_order_submit_for_signature_anonymous(self):
         Anonymous user should not be able to submit for signature an order.
         """
         order = factories.OrderFactory(
-            product=factories.ProductFactory(),
+            product=factories.ProductFactory(
+                contract_definition=factories.ContractDefinitionFactory()
+            ),
```

Comment:
Could be simplified to:
```suggestion
            product__contract_definition=factories.ContractDefinitionFactory(),
```